torch/csrc/distributed/autograd/utils.cpp

```diff
@@ -116,23 +116,23 @@ std::shared_ptr<FutureMessage> sendMessageWithAutograd(
     const WorkerInfo& dst,
     torch::distributed::rpc::Message&& wrappedRpcMsg,
     bool forceGradRecording,
-    const std::shared_ptr<torch::autograd::profiler::RecordFunction>& rf) {
+    const std::shared_ptr<torch::autograd::profiler::RecordFunctionAsync>& rf) {
   auto msg = getMessageWithAutograd(
       dst.id_,
       std::move(wrappedRpcMsg),
       MessageType::FORWARD_AUTOGRAD_REQ,
       forceGradRecording);
 
+  if (rf != nullptr) {
+    rf->exitScope();
```

Comment:
Question, why doing this before the send? in this case will the time cost of the send be accumulated to the outer record_function context?

Reply:
Sorry, maybe I should name `exitScope` right now. This doesn't actually stop the profiling so the time cost of the send will still be accumulated into this recordFunction (i.e. the `rf` object).

`exitScope` just resets some thread local state that's used for scoping management.